src/app/clusters/icd-management-server/icd-management-server.cpp

```diff
@@ -69,61 +71,203 @@ CHIP_ERROR IcdManagementAttributeAccess::Read(const ConcreteReadAttributePath &
 
 CHIP_ERROR IcdManagementAttributeAccess::ReadRegisteredClients(EndpointId endpoint, AttributeValueEncoder & encoder)
 {
-    return encoder.EncodeEmptyList();
+    uint16_t supported_clients = 0;
+    VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == Attributes::ClientsSupportedPerFabric::Get(endpoint, &supported_clients),
+                        CHIP_ERROR_NOT_FOUND);
+
+    return encoder.EncodeList([supported_clients](const auto & subEncoder) -> CHIP_ERROR {
+        IcdMonitoringEntry e;
+
+        const auto & fabricTable = Server::GetInstance().GetFabricTable();
+        for (const auto & fabricInfo : fabricTable)
+        {
+            PersistentStorageDelegate & storage = chip::Server::GetInstance().GetPersistentStorage();
+            IcdMonitoringTable table(&storage, fabricInfo.GetFabricIndex(), supported_clients);
+            for (uint16_t i = 0; i < table.Limit(); ++i)
+            {
+                CHIP_ERROR err = table.Get(i, e);
+                if (CHIP_ERROR_NOT_FOUND == err)
+                {
+                    return CHIP_NO_ERROR;
```

Comment:
Again: If we get CHIP_ERROR_NOT_FOUND here, why do we want to return CHIP_NO_ERROR?  This needs to be clearly documented.  Because as far as I can tell, this is _not_ the right behavior.

In which cases will CHIP_ERROR_NOT_FOUND happen here?